### PR TITLE
Help user avoid common movie/batch mistakes

### PR DESCRIPTION
**Pull request by PaulWessel**
All the **MOVIE_*** or **BATCH_*** script variables must have a leading $ or % to actually give a value.  It is pretty simple to forget to do so, and then you just have a random text in your script.  Now, both batch and movie looks for that. We also recogonize the deprecated ways to do a sub-shell (`...`) and recommend $(...) instead.
Finally, the common functions used by both batch and move were incorrectly named _gmtlib_.  They need to be _gmt__* per our convention.

### CODE REVIEW

File: src/gmt_support.c [hunk 1]
```diff
@@ -16975,13 +16975,46 @@ bool gmtlib_script_is_classic (struct GMT_CTRL *GMT, FILE *fp) {
 	return (!modern);
 }
 
+int gmt_token_check (struct GMT_CTRL *GMT, FILE *fp, char *prefix, unsigned int mode) {
```
Comment (seisman) [inline src/gmt_support.c:16978]: Some potential issues:
1. Tried with anim01, but it reports some errors like
```
Main script uses MOVIE_FRAME} but without the required leading $
```
2. Comment lines or inline comments should be skipped
3. csh doesn't support `$()` syntax

Comment (PaulWessel) [reply src/gmt_support.c:16978]: Addressed in the update

**Comment by seisman** [comment]
I'm not sure it's a good idea to check the MOVIE_* and BATCH_* variables. Sometimes users may want to write a string MOVIE_FRAME, e.g.,
```
echo "MOVIE_FRAME = ${MOVIE_FRAME}"
```

but this PR will report an error.

**Comment by PaulWessel** [comment]
Hm, I see you point.  But why would anyone do an echo in those scripts? Progress options are available.  I think the benefit of being told of problems outweigh at least that case you are presenting.

**Comment by seisman** [comment]
@PaulWessel This PR now fails, due to the renaming of some functions, e.g., gmt_set_comment.

**Pull request merged**